### ISSUE TITLE
Manual disconnect and reconnect to same URL

### ISSUE THREAD

**Issue by moehlone** [balderdashy/sails.io.js#43]
I want to disconnect manually from a socket and to a later time make a new connection to the same URL. At the moment it is not possible with sails.js - reconnection always fails.

Is there any possibility to solve this with sails.io.js?
See: https://github.com/Automattic/socket.io-client/issues/251

socket.io implemented in 1.0 a param called 'forceNew' for io.connect()

Best regards,
Phil

**Comment by moehlone** [comment]
more specific headline

**Comment by mattblang** [comment]
@moehlone I was just about to ask this same question.  In my case I want to unsubscribe from all models on log out, so I was hoping to just disconnect then reconnect the sails.io client.  I got it to disconnect using `io.socket.disconnect`, but I can't figure out how to `reconnect`.

**Comment by eulercodex** [comment]
I'm having the same issue as well.

**Comment by mattblang** [comment]
@eulercodex I used [location.reload](https://developer.mozilla.org/en-US/docs/Web/API/Location.reload) to accomplish this.  If you are doing a SPA and don't want the flicker of a refresh, pass `false` for `forceReload`.  The reload will happen from cache, but `Sails.io` will still reinitialize.  Personally, I was wanting to reset all of the subscriptions to the pubsub system on logout.

**Comment by eulercodex** [comment]
@mattblang  I don't particularly use sails. The issue was with socket.io but after looking at the doc online [version 1.2 release](http://socket.io/blog/socket-io-1-2-0/), I was able to solve my problem this way
```javascript
// on initialization
socket = io.connect({forceNew: true});
// for disconnection
socket.io.disconnect();
// for reconnection
socket = io.connect({forceNew: true});
```

**Comment by josuehenry** [comment]
```javascript
io.connect({forceNew: true});
```
works just fine

**Comment by josuehenry** [comment]
if you are using sails you need to edit sails.io.js SailsSocket function:

```javascript
   function SailsSocket (opts){
      var self = this;
      opts = opts||{};

      // Absorb opts
      self.useCORSRouteToGetCookie = opts.useCORSRouteToGetCookie;
      self.url = opts.url;
      self.multiplex = opts.multiplex;
      self.transports = opts.transports;

      // Absorb forceNew opts to force a new websocket connection every time
      self.forceNew = opts.forceNew;

      .
      .
      .
```

**Comment by muhammadghazali** [comment]
> if you are using sails you need to edit sails.io.js SailsSocket function:
> 
>    function SailsSocket (opts){
>       var self = this;
>       opts = opts||{};
> 
>       // Absorb opts
>       self.useCORSRouteToGetCookie = opts.useCORSRouteToGetCookie;
>       self.url = opts.url;
>       self.multiplex = opts.multiplex;
>       self.transports = opts.transports;
> 
>       // Absorb forceNew opts to force a new websocket connection every time
>       self.forceNew = opts.forceNew;
> 
>       .
>       .
>       .

@ghost @eulercodex I had the similar problem here, and seems this `forceNew` option does the job well...

**Comment by sailsbot** [comment]
Thanks for posting, @moehlone.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by twhtanghk** [comment]
The following way for power saving or reducing network traffic works fine in my case for platform browser or android.

```
# power saving or reduce network traffic
document.addEventListener 'pause', ->
	io.socket._raw.disconnect()
				
document.addEventListener 'resume', ->
	io.socket._raw.connect()
```
